sdk/sdk-core/aar/src/main/java/mil/emp3/core/services/kml/KMLSServiceListener.java

```diff
@@ -0,0 +1,38 @@
+package mil.emp3.core.services.kml;
+
+import android.util.Log;
+
+import java.util.concurrent.BlockingQueue;
+
+import mil.emp3.api.enums.KMLSEventEnum;
+import mil.emp3.api.events.KMLSEvent;
+import mil.emp3.api.exceptions.EMP_Exception;
+import mil.emp3.api.interfaces.IMap;
+import mil.emp3.api.listeners.IKMLSEventListener;
+
+/**
+ * Created by Matt.Miller on 9/29/2017.
+ */
+
+public class KMLSServiceListener implements IKMLSEventListener {
+    public static String TAG = KMLSServiceListener.class.getSimpleName();
+    IMap map;
+
+    BlockingQueue<KMLSEventEnum> queue;
+    public KMLSServiceListener(BlockingQueue<KMLSEventEnum> queue) {
```

Comment:
comments as well as you should create a copy of the queue.  That way your object cannot be manipulated by those who pass it to the constructor. Also use final in this class as well

Reply:
BlockingQueue? Make sure this is never called on the UI thread.